src/backends/beamcoder.ts

```diff
@@ -334,10 +330,17 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
                 // Beamcoder returns decoded packet frames as follows: [1000, 2000, 3000, 4000]
                 // If we're looking for a frame at 2500, we want to return the frame at 2000
                 const closestFrame = filteredFrames.reverse().find(f => f.pts <= targetPTS);
+
+                // The packet contains frames, but all of them have PTS larger than our a targetPTS (we looked too far)
                 if (!closestFrame) {
                     return outputFrame;
                 }
-                this.#filteredFrames = filteredFrames as beamcoder.DecodedFrames;
+
+                // store the filtered packet frames for later reuse
+                this.#filteredFramesPacket.unshift(filteredFrames);
+                if (this.#filteredFramesPacket.length > 2) {
+                    this.#filteredFramesPacket.pop();
```

Comment:
Nice, so we do clean up frames which are far away enough in the past!

Reply:
Yep, we only ever hold on the 2 packets worth of frames